<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
    <link rel="stylesheet" href="style.css">
</head>
<body>
    <!doctype html>
<html lang="pt-BR">
<head>
	<meta charset="utf-8">
	<title>Requisição XMLHttpRequest</title>
</head>
<body>
	<h1>Imagens Aleatórios de Cachorros</h1>
	<p>A partir do click no botão abaixo uma nova imagem aleatória de cachorros será carregada utilizando requisições assíncronas com XMLHttpRequest</p>

	<img id="img_dog" src="" alt="Aguardando a imagem ser carregada" />
	<br/>
	<button onclick="carregarImagens()">Carregar Imagens</button>
</body>
	<script type="text/javascript">

		function carregarImagens(){ // chamamos a função ao clicar no botão

			var url = "https://dog.ceo/api/breeds/image/random"; // API do json que estamos utilizando
			fetch(url, { 
				method: 'get' 
			}) // Realizando a requisição GET, mas não precisamos chama-la devido ser o padrão
			.then(function(response) { // then é usado para lidar com o resultado assíncrono da requisição. Response é o objeto que representa a resposta da requisição
				response.json().then(function(data){ // a resposta precisa ser convertida para JSON
                /*	O que é o then?
                    O then é usado com Promises (promessas).
                    Uma Promise representa um valor que estará disponível no futuro (por exemplo, depois que a requisição terminar).
                    O then permite que você execute código depois que a Promise é resolvida (quando a resposta chega
                */
                // Response.json() também retorna uma Promise, então usamos outro then para lidar com o JSON convertido
					console.log('Resultado da Requisição: ' + data.message); 

					var imgDog = document.getElementById("img_dog");
					imgDog.src = data.message;
				});
			})
			.catch(function(err) { 
				console.error('O seguinte erro ocorreu durante a requisição: ' + err);
			}); // catch err caso ocorra algum erro na requisição o codigo dentro do catch será executado

		}
	</script>
</html>
</body>
</html>